integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/api/RestApiAdminServiceTestCase.java

```diff
@@ -133,14 +235,25 @@ public void testDeleteAPI() throws Exception {
 
     @Test(groups = { "wso2.esb" },
           description = "Test API delete service for tenant ",
-          priority = 8)
+          priority = 15)
     public void testDeleteAPIForTenant() throws Exception {
         boolean apiResult = restAdminClient.deleteApiForTenant(tenantAPIName, tenantDomain);
         Assert.assertEquals(apiResult, true, "API was not removed");
 
     }
 
-    @AfterClass(groups = "wso2.esb")
+    @Test(groups = { "wso2.esb" },
+          description = "Test fault handling for empty API update ",
+          priority = 16,
+          expectedExceptions = RestApiAdminAPIException.class)
+    public void testAPIUpdateWithEmptyAPIName() throws Exception {
+        String emptyAPIName = "";
+        restAdminClient.updateAPIFromAPIData(emptyAPIName, api3);
+        Assert.fail("Expected exception not thrown for updating empty API");
+
+    }
+
+    @AfterClass(alwaysRun = true)
     public void close() throws Exception {
         if (restAdminClient.getAPICount() > 0) {
             restAdminClient.deleteAllApis();
```

Comment:
we should not call deleteAllApis. It will remove all api and lead to test failure in other classes

Reply:
Agree. Now that we are deploying all the artifacts initially it would be problematic for other test-cases. Updated with 8ea24440b5d2efb4a8940c1bd861c1a2f809c00f.